src/pages/home/report/ContextMenu/ContextMenuActions.js

```diff
@@ -96,20 +96,27 @@ export default [
         // the `text` and `icon`
         onPress: (closePopover, {reportAction, selection}) => {
             const message = _.last(lodashGet(reportAction, 'message', [{}]));
-            const html = lodashGet(message, 'html', '');
-
-            const parser = new ExpensiMark();
-            const reportMarkdown = parser.htmlToMarkdown(html);
-
-            const text = selection || reportMarkdown;
+            const messageHtml = lodashGet(message, 'html', '');
 
             const isAttachment = _.has(reportAction, 'isAttachment')
                 ? reportAction.isAttachment
                 : ReportUtils.isReportMessageAttachment(message);
             if (!isAttachment) {
-                Clipboard.setString(text);
+                // This error indicates new code that uses the selection object incorrectly.
+                if (selection && (selection.text || !selection.html)) {
+                    throw new Error('Selection has to either exclusively bear html or be null.');
+                }
```

Comment:
This is too restrictive, on the other hand, the code that sets the selection does not restrict the selection value. This is an open interface to pass any value into the selection. we should copy whatever is present in the selection. 
```suggestion
```
Instead set the text as HTML when only text is passed to allow both pastes to work. 

Why do you think we should restrict the selection to have HTML?

Reply:
This is the cleaned HTML under selection and it will be converted based on the platform. The selection has a generic signature for cases like link copying etc., which only uses `text`, but when we scrape the HTML under selection, we always pass it along. Here, in the case of comment selection, if we have the selection object without `html`, someone created a bug. We can change this to put something in the clipboard as a fallback as well as throw the error because this is simply an invalid state.

Reply:
@parasharrajat One other way that comes to mind is to revert the selection signature to string. Here, we expect HTML, so we transform the selection content to either MD or plaintext, depending on `Clipboard.canSetHtml`. That way, whatever is in the selection is put in the clipboard even if the output of the given transformation is the same as the input. This is the most unrestrictive catch-all solution that does not produce an error that I can think of.

Reply:
We can't say that if someone only sent a text, is a bug. We don't want to force anyone to send HTML. Both HTML and text are optional but one of them is a must. 

If we revert the signature of selection then others, then others won't have the option to send both HTML and text. We should clear our thoughts about selection. Now the selection is an object of `html and text` after this PR. IMO, keeping it that way throughout the app is a cleaner way. 

But yeah, that is also an option to revert the selection back to string and take it as HTML. 
 
> we transform the selection content to either MD or plaintext, depending on Clipboard.canSetHtml

I guess we are already doing that in the later part of the code but the early transformation is useless. we should delegate this until we know what we need as you have done in the later code.

Reply:
My motivation for `{html, text}` was to know when to throw this error (someone calls this differently/wrongly). The signature AFAIK becomes redundant if the different calls are valid. I'll push such a solution to overview.

Reply:
You are right that keeping html and text allows sending both, but should this ever happen given the text is never used in too many cases. I'll present a solution with the string signature and we can decide if we miss something.

Reply:
yeah, it will be great to see that as well. 

> , but should this ever happen given the text is never used in too many cases.

For now, it is only used for CopyToClipboard which requires both.

Reply:
Correct, but not if MD is needed, then plaintext is never used.

Reply:
That is only the case when Copying HTML fails which is very likely.